modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/pagemem/PageMemoryImpl.java

```diff
@@ -2319,92 +2317,46 @@ private long removePageForReplacement(PageStoreWriter saveDirtyPage) throws Igni
 
                     boolean pinned = PageHeader.isAcquired(absPageAddr);
 
-                    boolean skip = ignored != null && ignored.contains(rndAddr);
-
-                    if (relRmvAddr == rndAddr || pinned || skip) {
+                    if (pinned || ignored != null && ignored.contains(rndAddr) ||
+                        !electedPages.add(new PageWithAttrHolder(absPageAddr, rndAddr, fullId)) ||
+                        fullId.pageId() == storeMgr.metaPageId(fullId.groupId())) {
```

Comment:
You I checks 'fullId.pageId() == storeMgr.metaPageId(fullId.groupId())' only after page was added to electedPages, which is not correct because we should add to electedPages only valid pages. So you should swap condition or you should remove this page from electedPages if 'fullId.pageId() == storeMgr.metaPageId(fullId.groupId())' is true.